### ISSUE TITLE
Publish in Terraform.io registry to prepare to be available for terraform 0.13

### ISSUE THREAD

**Issue by sticky-note** [Telmate/terraform-provider-proxmox#179]
Provider binaries can be automatically hosted by Terraform registry as of MajorRelease 0.13:
cf. https://github.com/hashicorp/terraform/tree/guide-v0.13-beta/provider-sources

**Comment by sticky-note** [comment]
@ggongaware What do you think about this issue. It can be sufficient to do this and close #80 .

**Comment by sticky-note** [comment]
I opened an issue on Terraform repository because I can't find a way to install `proxmox` provider with actual implementation. cf. https://github.com/hashicorp/terraform/issues/25172

**Comment by paultyng** [comment]
👋 I work on the Providers / Registry team at HashiCorp, happy to help you all get this published as well.  Not sure if anyone has applied for this provider yet for access to publishing in the closed beta, but to get invited to the closed beta, please email terraform-registry-beta@hashicorp.com. We need:

- A list of GitHub usernames to add to the beta (you and any team members who will publish)
- List of provider repositories you plan to publish
- A GPG public key in ASCII-armor format, which you will be using to sign your provider releases

You can use one key for all of your providers, or separate keys if you prefer. If you are publishing from an organization, this key or keys will be associated with that namespace. Once in the beta, you can manage personal keys in the UI as well.

**Comment by sticky-note** [comment]
I'm actually interested to take part of this as I am currently testing this `provider` with terraform 0.13.0-beta1 to use it inside modules

**Comment by goffinf** [comment]
I agree this would be useful, but of course requires the provider and provisioner binaries to be compiled for each supported OS. I suspect you will know already but you can publish the Proxmox Terraform provider and provisioner to a local mirror. I use this approach and can confirm that it is successful, following the docs from HashiCorp here: https://www.terraform.io/docs/configuration/provider-requirements.html#in-house-providers

**Comment by paultyng** [comment]
We have a goreleaser.yml config file that will do all the cross compilation, summing, signing, for you, etc to create registry compatible releases, its referenced in our publishing docs: https://www.terraform.io/docs/registry/providers/publishing.html#creating-a-github-release

We also have a GitHub action that will automatically run GoReleaser for every `v*` tag: https://github.com/hashicorp/terraform-provider-scaffolding/blob/master/.github/workflows/release.yml

**Comment by srufle** [comment]
I added the plugin locally using information from 
https://learn.hashicorp.com/tutorials/terraform/provider-use

Changes your steps in **Copy executables to plugin directory** 
https://github.com/Telmate/terraform-provider-proxmox/blob/master/docs/installation.md

from ` ~/.terraform.d/plugins`
to ` ~/.terraform.d/plugins/telmate.com/tf/proxmox/0.1/linux_amd64`

Add section to main.tf
```
terraform {
    required_providers {
        proxmox = {
            source = "telmate.com/tf/proxmox"
        }
    }
}
```
Then `terraform init` succeeds

**Comment by robertminer1** [comment]
If you're running on Darwin (OSX) it will look like this:

```
$ cp bin/terraform-provider-proxmox ~/.terraform.d/plugins/telmate.com/tf/proxmox/0.1/darwin_amd64
$ cp bin/terraform-provisioner-proxmox ~/.terraform.d/plugins/telmate.com/tf/proxmox/0.1/darwin_amd64
```

**Comment by sticky-note** [comment]
Closed by #228
